server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -27,20 +35,43 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
 
 /**
  * Context used to fetch the {@code _source}.
  */
 public class FetchSourceContext implements Writeable, ToXContentObject {
+    public static final NodeFeature HIDE_VECTORS_IN_SOURCE = new NodeFeature("search.hide_vectors_in_source");
 
     public static final ParseField INCLUDES_FIELD = new ParseField("includes", "include");
     public static final ParseField EXCLUDES_FIELD = new ParseField("excludes", "exclude");
+    public static final ParseField INCLUDE_VECTORS = new ParseField("include_vectors");
+
+    public static final Boolean DEFAULT_INCLUDE_VECTORS = null;
+    public static final FetchSourceContext FETCH_SOURCE = new FetchSourceContext(
+        true,
+        Strings.EMPTY_ARRAY,
+        Strings.EMPTY_ARRAY,
+        DEFAULT_INCLUDE_VECTORS
+    );
+    public static final FetchSourceContext FETCH_SOURCE_WITH_VECTORS = new FetchSourceContext(
+        true,
+        Strings.EMPTY_ARRAY,
+        Strings.EMPTY_ARRAY,
+        true
+    );
+    public static final FetchSourceContext DO_NOT_FETCH_SOURCE = new FetchSourceContext(
+        false,
+        Strings.EMPTY_ARRAY,
+        Strings.EMPTY_ARRAY,
+        DEFAULT_INCLUDE_VECTORS
```

Comment:
Shouldn't this be `false`?

Reply:
No, because of the same logic I presented [in this thread](https://github.com/elastic/elasticsearch/pull/113418#discussion_r1778996590). Since we actively modify the `SourceFilter` returned by `FetchSourceContext#filter` to add vector field _exclusions_, we are actually adding filters when `includeVectors == false`. When `fetchSource` is false, we don't want to add any filters. 

In fact, the more technically value here is `true` for the reasons above, but we decided earlier on to handle `includeVectors == null` as "exclude semantic_text vectors, if they exist".

It's definitely confusing because the logic is inverted from what it seems like it should be on first read. Probably worth a comment.